### ISSUE TITLE
mooring lies below the ground

### ISSUE THREAD

**Issue by dankelley** [dankelley/mooring#12]
I decided to eliminate the problem of the stagnant shape going to the surface (snapshot 1) and so started fiddling with the graph generally.  I was drawing the ground on top of the mooring line, so I switched the order ... and now I see that the line is going *under* the ground (snapshot 2).  This goes beyond `app()`, to how the `x` and `z` values for mooring elements are computed.  I'll take a look at this today or tomorrow.

<details>

**Snapshot 1**

<img width="1022" alt="Screen Shot 2021-06-02 at 6 14 58 PM" src="https://user-images.githubusercontent.com/99469/120633285-59e40980-c440-11eb-9f3d-17e3e9ba225b.png">

**Snapshot 2**

<img width="1022" alt="Screen Shot 2021-06-03 at 7 48 21 AM" src="https://user-images.githubusercontent.com/99469/120633120-26a17a80-c440-11eb-8e9d-9ccd46b77b0c.png">

**Comment by richardsc** [comment]
Aha! 

I liked my theory about the drag causing tension that in turn could balance more weight from the line, but a sub-bottom mooring line probably isn't too realistic 😄

**Comment by dankelley** [comment]
Also, we get negative tension, by applying the formulae as I've devised them.   This is not a new fact (it's implied from the others) but I made a plot to show it.  I see two things to do in this "too heavy to stand upright" condistoin.

1. Just chop out everything at the bottom that has negative tension (instruments and wires alike), and carry on with the modified mooring.  But that doesn't help for a person (like me, given one of my goals) who wants to model ground lines.

2. Place the unsupported part of the mooring line along the bottom.

I'll need to think about the best plan, but number 2 seems best right now.  Any comments are welcome.

<details>



``` r
library("mooring")
source("~/git/mooring/R/mooring.R")
m <- mooring(anchor(depth=1000), wire(model="1/2in wire/jack", length=727), float(model="Kiel SFS40in"))
#> range(T): -77.669 320
md <- discretise(m, by=1)
mdk <- knockdown(md, u=0.5)
par(mfrow=c(1, 3))
#plot(md, which="tension", fancy=TRUE, showDepths=FALSE)
#plot(md, which="shape", fancy=TRUE)
plot(mdk, which="tension", fancy=TRUE, showDepths=FALSE)
plot(mdk, which="shape", fancy=TRUE)
plot(tension(md),z(md),pch=20,cex=1/3);abline(v=0)
```

![](https://i.imgur.com/KH1Emyt.png)

<sup>Created on 2021-06-03 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by dankelley** [comment]
I decided on option 2: string the line along the bottom.  This is done just by range-trimming the angle, using
```R
phi <- ifelse(phi>pi/2, pi/2, phi)
```
(where phi is the angle of an element to the vertical -- see the vignette on theory).

This yields as below.  I think I like it.

<details>

``` r
library("mooring")
source("~/git/mooring/R/mooring.R")
m <- mooring(anchor(depth=1000), wire(model="1/2in wire/jack", length=727), float(model="Kiel SFS40in"))
md <- discretise(m, by=1)
mdk <- knockdown(md, u=0.5)
par(mfrow=c(2, 2))
plot(md, which="tension", fancy=TRUE, showDepths=FALSE)
plot(md, which="shape", fancy=TRUE)
plot(mdk, which="tension", fancy=TRUE, showDepths=FALSE)
plot(mdk, which="shape", fancy=TRUE)
```

![](https://i.imgur.com/SCbBYRw.png)

<sup>Created on 2021-06-03 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by dankelley** [comment]
Fixed in "develop" branch, commit 6027a9c69381c563d58ed35f6e68ebad196b1c8a
